arangod/RestHandler/RestVersionHandler.cpp

```diff
@@ -61,27 +73,24 @@ RestStatus RestVersionHandler::execute() {
   if (allowInfo) {
     result.add("version", VPackValue(ARANGODB_VERSION));
 
-    bool found;
-    std::string const& detailsStr = _request->value("details", found);
-    if (found && StringUtils::boolean(detailsStr)) {
-      result.add("details", VPackValue(VPackValueType::Object));
-      Version::getVPack(result);
+    if (includeDetails) {
+      addVersionDetails(server, result);
+    }
+  }
+  result.close();
+}
 
-      auto& serverFeature = server().getFeature<ServerFeature>();
-      result.add("mode", VPackValue(serverFeature.operationModeString()));
-      auto serverState = ServerState::instance();
-      if (serverState != nullptr) {
-        result.add("role", VPackValue(ServerState::roleToString(serverState->getRole())));
-      }
+RestStatus RestVersionHandler::execute() {
+  VPackBuilder result;
 
-      std::string host = ServerState::instance()->getHost();
-      if (!host.empty()) {
-        result.add("host", VPackValue(host));
-      }
-      result.close();
-    }  // found
-  }    // allowInfo
-  result.close();
+  ServerSecurityFeature& security = server().getFeature<ServerSecurityFeature>();
+
+  bool const allowInfo = security.canAccessHardenedApi();
+  bool found;
+  std::string const& detailsStr = _request->value("details", found);
+  bool const includeDetails = found && StringUtils::boolean(detailsStr);
```

Comment:
While we are at this anyway, let's simplify it to
```suggestion
  bool const includeDetails = _request->parsedValue("details", false);
```

Reply:
Nice!